BNNApproximation.py

```diff
@@ -35,15 +35,21 @@ def sample(self, x, n):
         if self.dimensions.shape != s.shape or not a <= self.num_actions:
             raise ValueError("Invalid value to sample", s, a)
 
-        input_vector = np.concatenate((s, self.convert_action(a)))
+        input_vector = np.concatenate((self.convert_state(s), self.convert_action(a)))
 
-        # with self.session as sess:
-            # sess.run(tf.global_variables_initializer())
-            # sess.run(tf.tables_initializer())
-        sample_fn = self.bnn(np.array([[input_vector]], dtype=np.float32))
-        predictions = [self.session.run(sample_fn) for i in range(n)]
+
+        predictions = [self.bnn.predict(np.array([[input_vector]], dtype=np.float32)) for i in range(n)]
         return predictions
 
+    def convert_state(self, s):
```

Comment:
What is `s` here? Is it the list of indices from the tile coder: something like `[1, 29, 83, 185]`, or is it a sparse vector already? like: `[0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 1, ...]. Or perhaps it is from the tabular gridworld and `s` is an `(x, y)` coordinate. I'm realizing it is probably this last one.

I think that we should only be using this with the cts-gridworld domain, so `s` should only ever be a sparse vector. If we want to utilitize it in combination with the tabular stuff, we'll need to write a small compatibility layer (like this one) to go between them, but we won't want that compatibility layer to be the default.

Reply:
That's a good point. Should the interface then also expect a to be a sparse vector as well?